status/wallet2/network.nim

```diff
@@ -20,8 +20,8 @@ proc `$`*(self: Network): string =
 
 proc toPayload*(self: Network): JsonNode =
   return %* [{
-    "chain_id": self.chainID, "native_currency_decimals": self.nativeCurrencyDecimals, "layer": self.layer,
-    "chain_name": self.chainName, "rpc_url": self.rpcURL, "block_explorer_url": self.blockExplorerURL,
-    "icon_url": self.iconURL, "native_currency_name": self.nativeCurrencyName,
-    "native_currency_symbol": self.nativeCurrencySymbol, "is_test": self.isTest, "enabled": self.enabled
+    "chainId": self.chainID, "nativeCurrencyDecimals": self.nativeCurrencyDecimals, "layer": self.layer,
+    "chainName": self.chainName, "rpcUrl": self.rpcURL, "blockExplorerUrl": self.blockExplorerURL,
+    "iconUrl": self.iconURL, "nativeCurrencyName": self.nativeCurrencyName,
+    "nativeCurrencySymbol": self.nativeCurrencySymbol, "isTest": self.isTest, "enabled": self.enabled
   }]
```

Comment:
If you've gotten it to be unmarshalled correctly on the status-go side, you could simply do:
```suggestion
return %* [Json.encode(self).parseJson]
```
Or, create a generic implementation of `toPayload` that does this for any object:
```nim
proc toPayload*[T](self: T): JsonNode =
  return %* [Json.encode(self).parseJson]
```

Unrelated side note: it's pretty wasteful to convert to a `JsonNode`, only to convert back to a string when sending off to the RPC server. A better solution may be a generic proc for `callPrivateRpc` that accepts any object type for the payload and encodes it before sending.

Reply:
It worked by default on status-go, I was probably doing something totally wrong :) I updated the to payload as you suggested.

I think you are right we should update callPrivateRpc to accept an object which reply to toPayload ;)